### PR TITLE
chore: stricter mypy

### PR DESCRIPTION
**Pull request by henryiii**
I was looking at updating #1873 (and I realized it would be better to wait till after #1912) and I noticed we weren't very sensitive to updating things like `util.call` -> `utils.cmd.call` in our tests; I'd expect mypy to be able to detect all these without wasting time to run things. I also noticed this in #2048, where the tests weren't noticing the change to `enable`. These improvements should get proper mypy "coverage" of our tests. We still don't have 100% typing for arguments, but things like not allowing untyped function calls should require helper functions fully typed.

- **chore: improve mypy** (setuptools, pytest added, simpler strict setting, dropped unreachable since it depends on platform)
- **chore(types): type functions in tests**
- **chore(types): No partial types in tests**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -32,6 +32,7 @@ repos:
       - nox
       - packaging
       - pygithub
+      - pytest
```
Comment (Czaki) [inline .pre-commit-config.yaml:35]: Does having so many non pinned dependencies do not make pre-commit mypy fragile?

**Review by Czaki — APPROVED** [review]
Looks good

File: .pre-commit-config.yaml (hunk 1 above)
Comment (henryiii) [reply .pre-commit-config.yaml:35]: A little (though pytest isn't too bad). Though generally if something changes we just adapt to the change.

Pre-commit doesn't have a way to pin these. Maybe eventually uv will.

Comment (Czaki) [reply .pre-commit-config.yaml:35]: > Pre-commit doesn't have a way to pin these. Maybe eventually uv will.

What did you mean by `uv` will? pre-commit.ci do not allow to pass any constraints. 

The only possible way that I see is to have own CI job to bump constraints.

Comment (henryiii) [reply .pre-commit-config.yaml:35]: I mean if uv adds multiple independent locked environments (and tasks), we could use that to move this check to uv.

Comment (Czaki) [reply .pre-commit-config.yaml:35]: in napari (https://github.com/napari/napari) we use uv to compile a constraints file and call it from tox. 

Maybe the same approach could be used here?

Comment (henryiii) [reply .pre-commit-config.yaml:35]: If this becomes too much of a problem, we could probably do something (setting it up with nox would be easy). But for now it's fine, and long term we might end up moving from mypy to red-knot anyway.

**Pull request merged**